src/elements/forms/text-field.css

```diff
@@ -64,12 +78,53 @@ label {
   position: relative;
 }
 
+/* Helper message */
+
+.helper {
+  display: block;
+  padding-right: var(--form-control-padding-x);
+  padding-left: var(--form-control-padding-x);
+  margin-top: calc(1.5rem - 1em);
+  font-size: var(--form-helper-font-size);
+  color: var(--form-helper-color);
+
+  &:empty {
+    display: none;
+  }
+}
+
+input:invalid,
+input:invalid + label {
+  --form-control-color: var(--danger-dark);
+
+  /* Form some reason --form-control-color change is not enough */
+  --form-label-color: var(--form-control-color);
+}
```

Comment:
I do not think we should have this as a default behaviour. CSS validates during the input, JS validates on blur or submit. It's inconsistent and we should choose.

Since we develop a library, choosing ourselves would be opinionated what is the wrong approach for a library that pretends to be flexible (more or less) unless it's Prettier 😃